File: microservices/product/service/src/test/java/fr/edjaz/microservices/core/product/MapperTests.java
package fr.edjaz.microservices.core.product;

import fr.edjaz.api.core.product.Product;
import fr.edjaz.microservices.core.product.persistence.ProductEntity;
import fr.edjaz.microservices.core.product.services.ProductMapper;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;

import static org.junit.jupiter.api.Assertions.*;


class MapperTests {

    private final ProductMapper mapper = Mappers.getMapper(ProductMapper.class);

    @Test
    void mapperTests() {

        assertNotNull(mapper);

        Product api = new Product(1, "n", 1, "sa");

        ProductEntity entity = mapper.apiToEntity(api);

        assertEquals(api.getProductId(), entity.getProductId());
        assertEquals(api.getProductId(), entity.getProductId());
        assertEquals(api.getName(), entity.getName());
        assertEquals(api.getWeight(), entity.getWeight());

        Product api2 = mapper.entityToApi(entity);

        assertEquals(api.getProductId(), api2.getProductId());
        assertEquals(api.getProductId(), api2.getProductId());
        assertEquals(api.getName(),      api2.getName());
        assertEquals(api.getWeight(),    api2.getWeight());
        assertNull(api2.getServiceAddress());
    }
}

File: microservices/product/service/src/test/kotlin/fr/edjaz/microservices/core/product/MapperTests.kt
package fr.edjaz.microservices.core.product

import fr.edjaz.api.core.product.Product
import fr.edjaz.microservices.core.product.services.ProductMapper
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mapstruct.factory.Mappers

class MapperTests {
    private val mapper = Mappers.getMapper(ProductMapper::class.java)
    @Test
    fun mapperTests() {
        Assertions.assertNotNull(mapper)
        val api = Product(1, "n", 1, "sa")
        val entity = mapper.apiToEntity(api)
        Assertions.assertEquals(api.productId, entity.productId)
        Assertions.assertEquals(api.productId, entity.productId)
        Assertions.assertEquals(api.name, entity.name)
        Assertions.assertEquals(api.weight, entity.weight)
        val (productId, name, weight, serviceAddress) = mapper.entityToApi(entity)
        Assertions.assertEquals(api.productId, productId)
        Assertions.assertEquals(api.productId, productId)
        Assertions.assertEquals(api.name, name)
        Assertions.assertEquals(api.weight, weight)
        Assertions.assertNull(serviceAddress)
    }
}
